spanner/google/cloud/spanner/db_api/cursor.py

```diff
@@ -0,0 +1,114 @@
+import collections
+
+from google.cloud.spanner_v1.transaction import Transaction
+from google.cloud.spanner.db_api import Connection, exceptions
+from google.api_core.exceptions import InvalidArgument
+
+
+# Per PEP 249: A 7-item sequence containing information describing one result
+# column. The first two items (name and type_code) are mandatory, the other
+# five are optional and are set to None if no meaningful values can be
+# provided.
+Column = collections.namedtuple(
+    "Column",
+    [
+        "name",
+        "type_code",
+        "display_size",
+        "internal_size",
+        "precision",
+        "scale",
+        "null_ok",
+    ],
+)
+
+
+class Cursor(object):
+
+    def __init__(self, connection: Connection):
+        self.connection = connection
+        self.description = None
+        # Per PEP 249: The attribute is -1 in case no .execute*() has been
+        # performed on the cursor or the rowcount of the last operation
+        # cannot be determined by the interface.
+        self.rowcount = -1
+        # Per PEP 249: The arraysize attribute defaults to 1, meaning to fetch
+        # a single row at a time.
+        self.arraysize = 1
```

Comment:
I regret actually following the PEP for this case. In most cases (in BigQuery at least), they'll get the best performance if the default is `None`.

Reply:
Did what you suggested in 6284cdaf9ca6d1a241f625cb5071b27ee6af0770